packages/tracing/test/browser/browsertracing.test.ts

```diff
@@ -0,0 +1,231 @@
+import { BrowserClient } from '@sentry/browser';
+import { Hub } from '@sentry/hub';
+// tslint:disable-next-line: no-implicit-dependencies
+import { JSDOM } from 'jsdom';
+
+import { BrowserTracing, BrowserTracingOptions, getMetaContent } from '../../src/browser/browsertracing';
+import { DEFAULT_IDLE_TIMEOUT, IdleTransaction } from '../../src/idletransaction';
+
+let mockChangeHistory: ({ to, from }: { to: string; from?: string }) => void = () => undefined;
+let addInstrumentationHandlerType: string = '';
+
+jest.mock('@sentry/utils', () => {
+  const actual = jest.requireActual('@sentry/utils');
+  return {
+    ...actual,
+    addInstrumentationHandler: ({ callback, type }: any): void => {
+      addInstrumentationHandlerType = type;
+      mockChangeHistory = callback;
+    },
+  };
+});
+
+beforeAll(() => {
+  const dom = new JSDOM();
+  // @ts-ignore
+  global.document = dom.window.document;
```

Comment:
This is super ugly, know a better way around it?

Reply:
In test ugliness is OK